src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -145,30 +132,43 @@ - (instancetype)initWithRequestOptions:(GRPCRequestOptions *)requestOptions
 
   if ((self = [super init])) {
     _requestOptions = [requestOptions copy];
-    if (callOptions == nil) {
-      _callOptions = [[GRPCCallOptions alloc] init];
+    _callOptions = [callOptions copy];
+    if (!_callOptions) {
+      _actualCallOptions = [[GRPCCallOptions alloc] init];
     } else {
-      _callOptions = [callOptions copy];
+      _actualCallOptions = [callOptions copy];
```

Comment:
`_actualCallOptions = _callOptions` ?

Reply:
No we don't want them to be the same. It should be a copy.

Reply:
It seems you already copied, `_callOptions ` is the copy, otherwise `_callOptions ` is not used anymore.

Reply:
Correct, but `_callOptions` is a different copy from `_actualCallOptions`. It was intended to keep two different copied.